src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1040 @@
+            visualization: PropertyVisualization.Fill_Cell,
+            color: GREEN,
+            aux: 6n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: BLUE,
+            aux: 7n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: PURPLE,
+            aux: 8n,
+        },
+    ],
+    Color_By_Last_Digit_1: Array.from({length: 10}, (_item, index) => ({
+        property: Property.Last_Digit_Is,
+        visualization: PropertyVisualization.Fill_Cell,
+        color: RAINBOW[index],
+        aux: BigInt(index),
+    })),
+    Color_By_Last_Digit_2: Array.from({length: 10}, (_item, index) => ({
+        property: Property.Last_Digit_Is,
+        visualization:
+            index % 2
+                ? PropertyVisualization.Box_In_Cell
+                : PropertyVisualization.Fill_Cell,
+        color: RAINBOW[index],
+        aux: BigInt(index),
+    })),
+}
+
+function getPropertyParams(index: number, prop: PropertyObject) {
+    return {
+        [`property${index}`]: {
+            value: prop.property,
+            from: Property,
+            displayName: `Property ${index + 1}`,
+            required: false,
+            visibleDependency: index > 0 ? `property${index - 1}` : '',
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Vis`]: {
+            value: prop.visualization,
+            from: PropertyVisualization,
+            displayName: 'Display',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Color`]: {
+            value: prop.color,
+            forceType: 'color',
+            displayName: 'Color',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Aux`]: {
+            value: prop.aux,
+            displayName: (d: Property) => propertyAuxName[Property[d]] || '',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => Property[d] in propertyAuxName,
+        },
+    }
+}
+
+/*
+ *   FUNCTIONS TO CHECK FOR PROPERTIES
+ */
+function isPrime(factors: Factorization): boolean {
+    if (factors === null) throw new Error('Internal error in Grid')
+    if (factors.length == 0) return false // 1 is not prime
+    const [factor, power] = factors[0]
+    switch (factors.length) {
+        case 1:
+            return factor > 1n && power === 1n
+        case 2:
+            return factor === -1n && factors[1][1] === 1n // negative of prime
+    }
+    return false // two or more prime factors
+}
+
+//Adapted from Geeks For Geeks :
```

Comment:
I will fix these comments so there is a space after the `//`.

Reply:
Done in https://github.com/numberscope/frontscope/pull/230/commits/9cddb1cbd00ffae47823820f576cc37b78e76f91.